core/runtime/src/main/java/io/quarkus/runtime/QuarkusBindException.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.runtime;
+
+import java.net.BindException;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An exception that is meant to stand in for {@link BindException} and provide information
+ * about the target which caused the bind exception.
+ */
+public class QuarkusBindException extends BindException {
+
+    private final List<Integer> ports;
+
+    public QuarkusBindException(int port) {
+        this(Collections.singletonList(port));
+    }
+
+    public QuarkusBindException(List<Integer> ports) {
+        if (ports.isEmpty()) {
+            throw new IllegalStateException("ports must not be empty");
```

Comment:
Hum, this is surely very unexpected, but wouldn't it be confusing for the user if he got that instead of e.g. `Port ? seems to be in use by another process.` (note the `?` instead of the actual port)?
Might be better to add this as a suppressed exception?

Reply:
I just realized that suppression doesn't work as the exception is not (fully) logged. So maybe just a warning?
IDK, really just a minor thing...

Reply:
The reason I added this check is that I wanted to guard against erroneous usage and not have to deal with empty ports when inspecting the exception

Reply:
Sure, that makes sense. But if it really kicks in and a user is faced with it, then it will be more puzzling.

I guess in the end it's more of a "quarkus developer" guard than anything that will ever pop up for an actual quarkus user.

Reply:
Yeah, exactly! It's meant to be used by Quarkus developers (as it's part of the runtime package, so not meant to be part of the public API)